VideoWeb/VideoWeb/ClientApp/src/app/app.component.spec.ts

```diff
@@ -70,47 +90,63 @@ describe('AppComponent', () => {
+        locationSpy = jasmine.createSpyObj<Location>('Location', ['back']);
     });
 
-    beforeEach(() => {
-        component = new AppComponent(
-            routerSpy,
-            deviceTypeServiceSpy,
-            profileServiceSpy,
-            errorServiceSpy,
-            titleServiceSpy,
-            activatedRoute,
-            connectionStatusServiceSpy,
-            pageTrackerServiceSpy,
-            testLanguageServiceSpy,
-            translateServiceSpy,
-            oidcSecurityService,
-            configServiceSpy,
-            publicEventsServiceSpy,
-            new MockLogger()
-        );
-
-        document.getElementById = jasmine.createSpy('HTML Element').and.returnValue(dummyElement);
-        component.main = new ElementRef(dummyElement);
-        component.skipLinkDiv = new ElementRef(dummyElement);
-        deviceTypeServiceSpy.isSupportedBrowser.and.returnValue(true);
-        routerSpy.navigate.and.returnValue(Promise.resolve(true));
-        routerSpy.navigateByUrl.and.returnValue(Promise.resolve(true));
-        routerSpy.navigate.calls.reset();
-        routerSpy.navigateByUrl.calls.reset();
-        profileServiceSpy.getUserProfile.calls.reset();
-    });
+    beforeEach(
+        waitForAsync(() => {
+            TestBed.configureTestingModule({
+                providers: [
+                    { provide: DeviceTypeService, useValue: deviceTypeServiceSpy },
+                    { provide: ProfileService, useValue: profileServiceSpy },
+                    { provide: ErrorService, useValue: errorServiceSpy },
+                    { provide: Title, useValue: titleServiceSpy },
+                    { provide: ConnectionStatusService, useValue: connectionStatusServiceSpy },
+                    { provide: PageTrackerService, useValue: pageTrackerServiceSpy },
+                    { provide: TestLanguageService, useValue: testLanguageServiceSpy },
+                    { provide: TranslateService, useValue: translateServiceSpy },
+                    { provide: OidcSecurityService, useValue: oidcSecurityService },
+                    { provide: ConfigService, useValue: configServiceSpy },
+                    { provide: PublicEventsService, useValue: publicEventsServiceSpy },
+                    { provide: Location, useValue: locationSpy },
+                    { provide: Logger, useValue: new MockLogger() },
+                    { provide: Router, useValue: routerSpy },
+                    {
+                        provide: ActivatedRoute,
+                        useValue: activatedRouteMock
+                    }
+                ],
+                declarations: [
+                    AppComponent,
+                    TranslatePipeMock,
+                    MockComponent(BackNavigationComponent),
+                    MockComponent(HeaderComponent),
+                    MockComponent(FooterComponent),
+                    MockComponent(BetaBannerComponent)
```

Comment:
I've used this package a lot in previous projects. I find it helps hugely with UnitTests and keeping the tests focused on the specific component. I would recommend using this going forward.